controllers/storageconsumer/storageconsumer_upgrade_controller.go

```diff
@@ -143,12 +147,34 @@ func (r *StorageConsumerUpgradeReconciler) Reconcile(ctx context.Context, reques
 		cephFsClaimName := util.GenerateNameForCephFilesystemStorageClass(storageCluster)
 		cephFsClaimMd5Sum := md5.Sum([]byte(cephFsClaimName))
 		cephFSClientProfile := hex.EncodeToString(cephFsClaimMd5Sum[:])
-		cephFsStorageRequestHash := getStorageRequestName(string(storageConsumer.UID), cephFsClaimName)
+		cephFsStorageRequestHash := getStorageRequestHash(string(storageConsumer.UID), cephFsClaimName)
 		cephFsNodeSecretName := storageClaimCephCsiSecretName("node", cephFsStorageRequestHash)
 		cephFsProvisionerSecretName := storageClaimCephCsiSecretName("provisioner", cephFsStorageRequestHash)
-		cephFsStorageRequestMd5Sum := md5.Sum([]byte(cephFsStorageRequestHash))
+		cephFsStorageRequestName := getStorageRequestName(string(storageConsumer.UID), cephFsClaimName)
+		cephFsStorageRequestMd5Sum := md5.Sum([]byte(cephFsStorageRequestName))
 		svgName := fmt.Sprintf("cephfilesystemsubvolumegroup-%s", hex.EncodeToString(cephFsStorageRequestMd5Sum[:16]))
 
+		//Delete the ownerRef on the resources svg
+		svg := &metav1.PartialObjectMetadata{}
+		svg.SetGroupVersionKind(rookCephv1.SchemeGroupVersion.WithKind("CephFilesystemSubVolumeGroup"))
+		svg.Name = svgName
+		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(storageConsumer), svg); client.IgnoreNotFound(err) != nil {
+			return reconcile.Result{}, err
+		}
+		if svg.UID != "" {
+			refs := svg.GetOwnerReferences()
+			if idx := slices.IndexFunc(refs, func(owner metav1.OwnerReference) bool {
+				return owner.Kind == "StorageRequest"
+			}); idx != -1 {
+				svg.SetOwnerReferences(slices.Delete(refs, idx, idx+1))
+			}
+		}
```

Comment:
- Why is this code located here? it has nothing to do with filling in the CM 
- I don't see any code that applies the change. As the code stands now this is only an in memory change.